### DISCUSSION TITLE
Merge OpenGL and Cairo Mobjects

### DISCUSSION THREAD

**Discussion by awesome-developer-20** [ManimCommunity/manim#1579] [Architecture]
Problem:
Maintaining two separate mobject trees is difficult and from user end to changing all constructor names from `Tex` to `OpenGLTex` is irritating to change rendering method.

Solution:
Merge `Mobject` and `OpenGLMobject` classes and separate out rendering logic in renderer.

**Comment by behackl** [comment]
We are already in the process of doing exactly that, via the metaclass approach introduced in #1272. :-)

**Reply by awesome-developer-20** [reply]
https://github.com/ManimCommunity/manim/pull/1577#issuecomment-849421550

> I'd still like to process the remaining mobjects that currently inherit from VMobject explicitly on a module-by-module basis rather than just enabling it everywhere by adding the metaclass to VMobject.
> 
> My suggestion for future work would be:
> 
> 1. Process the remaining modules that have not been OpenGL-ified yet,
> 2. Make the implementations of VMobject and OpenGLVMobject compatible, then move the metaclass up one layer,
> 3. then investigate Mobject vs OpenGLMobject.

This will be great.

Just one question, eventually you all are aiming to remove metaclasses and merge opengl and cairo mobject classes, Right?

**Reply by kilacoda-old** [reply]
We're planning to move away from cairo entirely in the near future.

**Reply by awesome-developer-20** [reply]
> We're planning to move away from cairo entirely in the near future.

That really awesome!!

**Reply by awesome-developer-20** [reply]
Till when you are planning to completely move to opengl?

**Reply by behackl** [reply]
We don't have a concrete timeframe.

**Reply by awesome-developer-20** [reply]
It's okay, Thanks
